File: Engineering concepts/SingletonPatternExample/TestLogger.java
public class TestLogger {

    public static void main(String[] args) {

        System.out.println("===== Singleton Pattern Demo =====\n");

        Logger logger1 = Logger.getInstance();
        Logger logger2 = Logger.getInstance();

        logger1.log("Application Started");
        logger2.log("User Logged In");

        System.out.println();

        if (logger1 == logger2) {
            System.out.println(" Only one Logger instance exists.");
        } else {
            System.out.println("Multiple Logger instances exist.");
        }

        System.out.println("\nProgram Executed Successfully.");
    }
}

File: Deep Skilling/Engineering concepts/Design Patterns and Principles/SingletonPatternExample/TestLogger.cs
using System;

namespace SingletonPatternExample
{
    class TestLogger
    {
        static void Main(string[] args)
        {
            Console.WriteLine("===== Singleton Pattern Demo =====\n");

            Logger logger1 = Logger.GetInstance();
            Logger logger2 = Logger.GetInstance();

            logger1.Log("Application Started");
            logger2.Log("User Logged In");

            Console.WriteLine();

            if (object.ReferenceEquals(logger1, logger2))
            {
                Console.WriteLine(" Only one Logger instance exists.");
            }
            else
            {
                Console.WriteLine("Multiple Logger instances exist.");
            }

            Console.WriteLine("\nProgram Executed Successfully.");
        }
    }
}
